aten/src/TH/generic/THTensor.cpp

```diff
@@ -73,10 +74,11 @@ THTensor *THTensor_(newWithStorage1d)(THStorage *storage, ptrdiff_t storageOffse
                                int64_t size0, int64_t stride0)
 {
   c10::raw::intrusive_ptr::incref(storage);
```

Comment:
Assumed invariant: dtype of storage matches `scalar_t`. Pretty sure this is true.

Reply:
Looks like all the callers of `THTensor_(newWithStorage1d)` are copy operations, and the dtypes should match in those cases.